aea/protocols/dialogue/base.py

```diff
@@ -1228,9 +1236,10 @@ def is_dialogue_present(self, dialogue_label: DialogueLabel) -> bool:
 
     def get_latest_label(self, dialogue_label: DialogueLabel) -> DialogueLabel:
         """Get latest label for dialogue."""
-        return self._incomplete_to_complete_dialogue_labels.get(
+        complete_dialogue_label = self._incomplete_to_complete_dialogue_labels.get(
             dialogue_label, dialogue_label
         )
+        return complete_dialogue_label if complete_dialogue_label else dialogue_label
```

Comment:
```suggestion
        return self._incomplete_to_complete_dialogue_labels.get(dialogue_label) or dialogue_label
```
we know key exists and if value is None we get back dialogue_label